pkg/util/url_test.go

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+package util
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestIsValidURL(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name     string
+		endpoint string
+	}{
+		{
+			name:     "valid url",
+			endpoint: "https://sts.eu-central-1.amazonaws.com",
+		},
+		{
+			name:     "invalid url no scheme",
+			endpoint: "sts.eu-central-1.amazonaws.com",
+		},
+		{
+			name:     "invalid url invalid scheme setup",
+			endpoint: "https:///sts.eu-central-1.amazonaws.com",
+		},
+		{
+			name:     "invalid url no host",
+			endpoint: "https://",
+		},
+	}
+
+	for i, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			valid := IsValidURL(test.endpoint)
+
+			if i == 0 {
```

Comment:
let's add a field in the test `struct`, which is `valid bool` and use that instead of the sequence of the test case. This allows to add new valid test cases later on without having to modify the core of the test and also keeps the information of  the test case in one place (only in the slice element)

Reply:
fixed. Good catch. Totally agree